src/Http/Offer/LegacyTimeSpansParser.php

```diff
@@ -16,22 +16,25 @@ public function parse(stdClass $data): StdClass
 
         // Rename timeSpans to subEvent
         if (isset($data->timeSpans) && is_array($data->timeSpans) && !isset($data->subEvent)) {
-            $data->subEvent = array_map(
-                function ($timeSpan) {
-                    // Rename start to startDate
-                    if ($timeSpan instanceof stdClass && isset($timeSpan->start)) {
-                        $timeSpan->startDate = $this->formatDateTime($timeSpan->start);
-                        unset($timeSpan->start);
-                    }
-                    // Rename end to endDate
-                    if ($timeSpan instanceof stdClass && isset($timeSpan->end)) {
-                        $timeSpan->endDate = $this->formatDateTime($timeSpan->end);
-                        unset($timeSpan->end);
-                    }
-                    return $timeSpan;
-                },
-                $data->timeSpans
-            );
+            if (count($data->timeSpans) > 0) {
```

Comment:
Nit: Can this `if` not be part of the `if` above it?

```
if (isset($data->timeSpans) && is_array($data->timeSpans) && !isset($data->subEvent) && count($data->timeSpans) > 0))
```

Reply:
Looks good otherwise though!

Reply:
I always wanted to do the `unset($data->timeSpans);`

Reply:
Ah I missed that! Good call.